migration-engine/connectors/sql-migration-connector/src/sql_schema_differ/differ_database.rs

```diff
@@ -249,6 +254,45 @@ impl<'a> DifferDatabase<'a> {
             .filter(move |previous| !self.next_enums().any(|next| enums_match(previous, &next)))
     }
 
+    pub(crate) fn schemas(&self) -> Pair<&'a SqlDatabaseSchema> {
+        self.schemas
+    }
+
+    /// Extensions not present in the previous schema.
+    pub(crate) fn created_extensions(&self) -> impl Iterator<Item = ExtensionId> + '_ {
+        self.next_extensions()
+            .filter(move |next| !self.previous_extensions().any(|prev| next.name() == prev.name()))
```

Comment:
This has quadratic complexity. We can make it linear by using `db.extensions`. Will do in an extra commit.